apps/homework/models.py

```diff
@@ -147,6 +149,20 @@ def score(self):
             self.retrieve_score_and_status()
         return self.stored_score
 
+    @property
+    def logs(self):
+        if self.stored_logs == None or len(self.stored_logs.keys()) == 0:
+            self.retrieve_score_and_status()
+        else:
+            # Try the first URL in our logs
+            sas_url = self.stored_logs[list(self.stored_logs.keys())[0]]
+            resp = requests.get(sas_url)
+            if not resp.ok:
+                logger.info("Submission SAS urls for logs appear expired, retrieving new ones.")
+                self.retrieve_score_and_status()
+        return self.stored_logs
```

Comment:
This looks like a solid long-term solution. If we have issues with performance in the future, caching would be a good option, I think.